packages/next/src/server/next-server.ts

```diff
@@ -1235,6 +1340,172 @@ export default class NextNodeServer extends BaseServer {
 
         const match = await this.matchers.match(pathname, options)
 
+        if (this.isRouterWorker) {
+          let page = pathname
+
+          if (!(await this.hasPage(page))) {
+            for (const route of this.dynamicRoutes || []) {
+              if (route.match(pathname)) {
+                page = route.page
+              }
+            }
+          }
```

Comment:
Is there a reason this couldn't use the RouteMatchers? They should already be conifgured with the dynamic routes that are available.

Additionally, this for loop (even when it finds a match) will go through every route, shouldn't there be a `break` here?